cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/connector/resource/AwsResourceVolumeConnector.java

```diff
@@ -0,0 +1,90 @@
+package com.sequenceiq.cloudbreak.cloud.aws.common.connector.resource;
+
+import static java.util.stream.Collectors.toMap;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.TimeUnit;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.dyngr.Polling;
+import com.dyngr.core.AttemptResults;
+import com.sequenceiq.cloudbreak.cloud.ResourceVolumeConnector;
+import com.sequenceiq.cloudbreak.cloud.aws.common.client.AmazonEc2Client;
+import com.sequenceiq.cloudbreak.cloud.aws.common.view.AuthenticatedContextView;
+import com.sequenceiq.cloudbreak.cloud.context.AuthenticatedContext;
+import com.sequenceiq.cloudbreak.cloud.model.CloudResource;
+import com.sequenceiq.cloudbreak.cloud.model.VolumeSetAttributes;
+import com.sequenceiq.cloudbreak.cloud.model.VolumeSetAttributes.Volume;
+
+import software.amazon.awssdk.services.ec2.model.DeleteVolumeRequest;
+import software.amazon.awssdk.services.ec2.model.DescribeVolumesRequest;
+import software.amazon.awssdk.services.ec2.model.DescribeVolumesResponse;
+import software.amazon.awssdk.services.ec2.model.DetachVolumeRequest;
+import software.amazon.awssdk.services.ec2.model.VolumeState;
+
+@Service
+public class AwsResourceVolumeConnector implements ResourceVolumeConnector {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AwsResourceVolumeConnector.class);
+
+    private static final int MAX_READ_COUNT = 15;
+
+    private static final int SLEEP_INTERVAL = 2;
+
+    @Override
+    public void detachVolumes(AuthenticatedContext authenticatedContext, List<CloudResource> cloudResources) throws Exception {
+        Map<String, List<Volume>> instanceVolumeIdsMap = cloudResources.stream().filter(res -> null != res.getInstanceId())
```

Comment:
Don't we have to filter for only Volume resource types here?

Reply:
detach volumes request requires both the instance id and volume id - that is the reason i chose Map<instance id, List<volumes>> not just volumes. Just volume doesn't have instance id on it.